### PR TITLE
Simplify join methods and add tests

### PR DESCRIPTION
**Pull request by edwardmlyte**
Add unit tests for helper method, and simplify the method.
Had to add check for separator not being present to avoid NullPointerException and added a unit test to cover this.

### CODE REVIEW

File: src/test/java/com/currencycloud/client/UtilsTest.java [hunk 1]
```diff
@@ -15,9 +23,35 @@ public void returnsRootCauseForChain() throws Exception {
...
+    }
+
+    @Test
+    public void joinInverse_nullList_nullReturned() {
+        String result = Utils.joinInverse(null, "|");
+        assertThat(result, is(nullValue()));
+    }
+
+    @Test
+    public void joinInverse_nullSeparator_nullReturned() {
```
Comment (edwardmlyte) [inline src/test/java/com/currencycloud/client/UtilsTest.java]: this is a change in behaviour. previously, `null` being fed in would mean a string of `null` would be used as a separator. perhaps a better behaviour would be a `null` separator lead to the strings being reversed and concatenated

File: src/main/java/com/currencycloud/client/Utils.java [hunk 2]
```diff
@@ -1,39 +1,33 @@
...
-            sb.append(string);
-        }
-        return sb.toString();
+
+        List<String> reversed = new ArrayList<>(strings);
+        Collections.reverse(reversed);
+        return join(reversed, separator);
+    }
+
+    private static boolean validInput(List<String> strings, String separator) {
```
Comment (jonathancouchman) [inline src/main/java/com/currencycloud/client/Utils.java]: would it not be better to have a test "invalidInputs"

`return strings == null || separator == null`

then you would not need to test !validInput(...), but rather invalidInputs(...)

File: src/main/java/com/currencycloud/client/Utils.java [hunk 3]
```diff
@@ -1,39 +1,33 @@
...
 import java.util.List;
 
 public enum Utils {
     ;
 
     public static final String dateFormat = "yyyy-MM-dd'T'HH:mm:ssX";
 
-    public static String join(Iterable<String> strings, String separator) {
-        if (strings == null) {
+    public static String join(List<String> strings, String separator) {
```
Comment (jonathancouchman) [inline src/main/java/com/currencycloud/client/Utils.java]: You are changing the signature of a public method that may be used by clients.

Comment (edwardmlyte) [reply src/main/java/com/currencycloud/client/Utils.java]: Good point. Makes more sense to make the other method use Iterable (which would be backwards compatible)

Comment (jonathancouchman) [reply src/main/java/com/currencycloud/client/Utils.java]: You can mark the old method deprecated if it is not being used internally

**Comment by edwardmlyte** [comment]
@EAddario honestly, this was just initial stuff to do before I had more access. However it highlights we're handcrafting unnecessary code and there's room for improvement on the unit test front too. Perhaps the two improvements could be done separately. First, add unit tests. Then simplify and demonstrate with the tests that the behaviour hasn't changed. Happy to do whatever.

**Comment by edwardmlyte** [comment]
Doesn't add features.

**Comment by EAddario** [comment]
Agree changes do not always need to add new functionality but they do always need to add value, be stability or functionality or testability. The proposed PR was valid but the use case(s) was unclear, hence the question.

**Pull request closed without merging**